### DISCUSSION TITLE
Well formed /api/v1/photos GET request yields HTTP 400 Bad Request "Unable to do that" with newly generated access token and HTTP 200 with browser generated access token.

### DISCUSSION THREAD

**Discussion by davidandreoletti** [photoprism/photoprism#4667] [Q&A] [answered]
#### 1. What is not working as documented?

An HTTP request to get photos using the access token gathered from my browser will get HTTP 200 server response, while the same request using an access token generated from `photoprism auth add --scope '*' --name 'Test' --expires -1` will get HTTP 400 "Unable to do that".  

#### 2. How can we reproduce it?

1. Logging as administrator on the Photoprism website on Google Chrome
2. Open the "Search" page
3. Open the browser "Inspector"
4. Refresh the webpage (to get the Inspector's "Network" tab populated) 
5. Get the X-Auth-Token value. Let's name it `valid-token` throughout this report.
6. Send `curl --verbose -X 'GET' 'http://example.com/api/v1/photos?count=10' -H 'X-Auth-Token: valid-token'` to obtain HTTP 200 and  the list of photos as payload.
7. Create an access token with `photoprism auth add --scope '*' --name 'Test' --expires -1`
8. Get the token value. Let's name it `problematic-token` throughout this report.
9. Send `curl --verbose -X 'GET' 'http://example.com/api/v1/photos?count=10' -H 'X-Auth-Token: problematic-token'` to obtain HTTP 400 and the error message "Unable to do that".

#### 3. What behavior do you expect?

Upon step 9, the list of photos to be returned (same as step 6).

#### 4. What could be the cause of your problem?

No idea as the HTTP request is well formed and step 9's token is generated using a [supported method](https://docs.photoprism.app/developer-guide/api/oauth2/#access-tokens).

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

N/A.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64
(b) Database Type & Version: SQLite
(c) Operating System Types & Versions: Linux
(d) Browser Types & Versions:, Chrome (Desktop)
(e) Ad Blockers, Browser Plugins, and/or Firewall Software: No ad blocker / plugins / firewall.
(f) Photoprism version: PhotoPrism® CE / Build [240915-e1280b2fb](https://docs.photoprism.app/release-notes/)

#### 7. On what kind of device is PhotoPrism installed?

Irrelevant.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

Reverse Proxy: Nginx

**Comment by graciousgrey** [comment, marked as answer]
This should work if you create a token that is assigned to a user:

https://docs.photoprism.app/user-guide/users/client-credentials/#app-passwords

**Reply by davidandreoletti** [reply]
Works now. Thanks.

**Comment by lastzero** [comment]
That's right: you can either use an app password as a token (which is always associated with a user account), or manually create an access token that is associated with a user account (e.g. `admin` if you want full access). Should you then still have problems, feel free to open a GitHub Discussion for further help using the API.
